### DISCUSSION TITLE
Navigation item looks bad

### DISCUSSION THREAD

**Discussion by DotCat1985** [OrchardCMS/OrchardCore#11315] [Q&A]
Hi,
I've enabled the dashboard menu item **FileManager** developing the following files:

**AdminMenu.cs**
```
using Microsoft.Extensions.Localization;
using OrchardCore.Navigation;
using System;
using System.Threading.Tasks;

namespace OrchardCore.FileManager
{
    public class AdminMenu : INavigationProvider
    {
        private readonly IStringLocalizer S;

        public AdminMenu(IStringLocalizer<AdminMenu> localizer)
        {
            S = localizer;
        }

        public Task BuildNavigationAsync(string name, NavigationBuilder builder)
        {
            if (String.Equals(name, "admin", StringComparison.OrdinalIgnoreCase))
            {
                builder
                    .Add(S["File manager"], NavigationConstants.AdminMenuWorkflowsPosition, fileManager => fileManager
                        .AddClass("filemanager").Id("filemanager")
                        .Action("Index", "Admin", new { area = "OrchardCore.FileManager" })
                            .Permission(Permissions.ManageFileManager)
                            .LocalNav());
                builder
                    .Add(S["Configuration"], configuration => configuration
                        .Add(S["Settings"], settings => settings
                            .Add(S["File manager"], S["File manager"].PrefixPosition(), settings => settings
                            .Action("Index", "Admin", new { area = "OrchardCore.Settings", groupId = FileManagerConstants.Features.FileManager })
                                .Permission(Permissions.ManageFileManager)
                                .LocalNav())
                            )
                        );
            }
            return Task.CompletedTask;
        }
    }
}
```

NavigationItemText-filemanager.Id.cshtml
`<span class="icon"><i class="fa fa-file" aria-hidden="true"></i></span><span class="title">@T["File manager"]</span>`

But, the navigation item looks bad (see the screenshot below).

![Screenshot (123)](https://user-images.githubusercontent.com/61869142/156900496-3f19a32b-1c54-4159-900d-fd70d9c4b324.png)

How can I fix this issue?
I'm developing `OrchardCore 1.2.2`.

**Comment by Skrypt** [comment]
This is not an admin menu item that comes from the source code.

**Reply by DotCat1985** [reply]
Sorry, but I didn't understand. Should I share another snippets or anything?

**Reply by Skrypt** [reply]
This "sort by" filter on the admin menu is not something that we did. This is custom code.

**Reply by DotCat1985** [reply]
> This "sort by" filter on the admin menu is not something that we did. This is custom code.

I'll explain well. The file manager UI component is rendered under the admin menu when I use the following snippet:

```
builder
                    .Add(S["File manager"], NavigationConstants.AdminMenuWorkflowsPosition, fileManager => fileManager
                        .AddClass("filemanager").Id("filemanager")
                        .Action("Index", "Admin", new { area = "OrchardCore.FileManager" })
                            .Permission(Permissions.ManageFileManager)
                            .LocalNav());
```

But, when I use the given snippet:

```
builder
                    .Add(S["File manager"], NavigationConstants.AdminMenuWorkflowsPosition, fileManager => fileManager
                        .Action("Index", "Admin", new { area = "OrchardCore.FileManager" })
                            .Permission(Permissions.ManageFileManager)
                            .LocalNav());
```
FileManager UI component i rendered under the title "File Manager".

**Reply by Skrypt** [reply]
Yes, there is a shape template associated with the `.Id("filemanage")` and there is a specific class associated with the menu item when you do `.AddClass("filemanager")` so this may be the issue there, it's not affecting the proper class on the menu items and breaks the CSS styles.

**Reply by DotCat1985** [reply]
> Yes, there is a shape template associated with the `.Id("filemanage")` and there is a specific class associated with the menu item when you do `.AddClass("filemanager")` so this may be the issue there, it's not affecting the proper class on the menu items and breaks the CSS styles.

OK, thank you.
